File: Domain/Workio.Domain/Entities/Application/Application.cs
using Workio.Common.Models.Entities;
using Workio.Common.Models.Enums;
using Workio.Domain.Entities.Identity;

namespace Workio.Domain.Entities.Application;

public class Application : BaseEntity
{
    public string Message { get; set; } = null!;
    public ApplicationStatus Status { get; set; } = ApplicationStatus.Pending;
    
    public Guid JobId { get; set; }
    public Guid UserId { get; set; }

    public User? User { get; set; }
    public Job.Job? Job { get; set; }
}

File: Presentation/Clients/template-web-ui/src/domain/application.ts
import { BaseEntity } from "./base/baseEntity";
import { Job } from "./job";

export interface Application extends BaseEntity {
  message: string;
  jobId: string;
  userId: string;
  status: ApplicationStatus;
  user: any;
  job: Job;
}

export enum ApplicationStatus {
  Pending,
  Accepted,
  Rejected,
}
